upload-api/functions/bridge.js

```diff
@@ -0,0 +1,331 @@
+}
+
+/**
+ * 
+ * @type {import('../bridge/types.js').BridgeReceiptBuilder}
+ */
+async function buildBridgeReceipts(receipts) {
+  /** @type {import('../bridge/types.js').BridgeReceipt[]} */
+  const bridgeReceipts = []
+  for (const receipt of receipts) {
+    if (receipt.root.data) {
+      bridgeReceipts.push({
+        data: receipt.root.data?.ocm,
+        sig: receipt.root.data?.sig
+      })
+    } else {
+      return {
+        error: {
+          name: 'BridgeReceiptFailure',
+          message: 'One of the receipts contains no data'
+        }
+      }
+    }
+  }
+  return { ok: bridgeReceipts }
+}
+
+/**
+ * @type {import('../bridge/types.js').BridgeBodyBuilder}
+ */
+function serializeBridgeReceipts(receipts) {
+  // TODO: use second parameter to pick serialization format
+  return dagJSON.stringify(receipts)
+}
+
+/**
+ * 
+ * @type {import('../bridge/types').TasksExecutor}
+ */
+export async function invokeAndExecuteTasks(
+  issuer, servicePrincipal, serviceURL, tasks, delegation
+) {
+  const invocations = tasks.map((task) => invoke({
+    issuer,
+    audience: servicePrincipal,
+    capability: {
+      can: task[0],
+      with: task[1],
+      nb: task[2]
+    },
+    proofs: [delegation]
+  }))
+  /**
+   * @type {import('@ucanto/interface').ConnectionView<import('@web3-storage/upload-api').Service>}
+   */
+  const connection = connect({
+    id: servicePrincipal,
+    codec: CAR.outbound,
+    channel: HTTP.open({
+      url: serviceURL,
+      method: 'POST'
+    })
+  })
+  // this is an annoying hack to make typescript happy - it wants at LEAST one argument and needs
+  // to be assured that that's the case
+  const [firstInvocation, ...restOfInvocations] = invocations
+  // @ts-ignore multiple issues here
+  return connection.execute(firstInvocation, ...restOfInvocations)
+}
+
+
+
+/**
+ * AWS HTTP Gateway handler for POST / with ucan invocation router.
+ *
+ * We provide responses in Payload format v2.0
+ * see: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format
+ *
+ * @param {import('aws-lambda').APIGatewayProxyEventV2} request
+ * @param {import('../bridge/types').BridgeRequestContext} context
+ */
+export async function handleBridgeRequest(request, context) {
+  try {
+    const { authSecretHeader, authorizationHeader, body, contentType } = parseAwsLambdaRequest(request)
+
+    if (!authorizationHeader) {
+      return {
+        statusCode: 401,
+        body: 'request has no Authorization header'
+      }
+    }
+
+    if (!authSecretHeader) {
+      return {
+        statusCode: 401,
+        body: 'request has no X-Auth-Secret header'
+      }
+    }
+
+    if (!body) {
+      return {
+        statusCode: 400,
+        body: 'request has no body'
+      }
+    }
+
+    if (!contentType) {
+      return {
+        statusCode: 400,
+        body: 'request has no Content-Type header'
```

Comment:
Personally I'd just assume `dag-json` if no header...

Reply:
works for me!